src/diffusers/quantizers/finegrained_fp8/finegrained_fp8_quantizer.py

```diff
@@ -0,0 +1,205 @@
+from typing import TYPE_CHECKING, Any, Dict, List, Optional
+
+from ...utils import get_module_from_name, is_accelerate_available, is_torch_available, logging
+from ..base import DiffusersQuantizer
+
+
+if is_torch_available():
+    import torch
+
+logger = logging.get_logger(__name__)
+
+if TYPE_CHECKING:
+    from ...models.modeling_utils import ModelMixin
+
+
+class FinegrainedFP8Quantizer(DiffusersQuantizer):
+    """
+    FP8 quantization implementation supporting both standard and MoE models.
+    Supports both e4m3fn formats based on platform.
```

Comment:
Can we expand on this a bit? What are both `e4m3fn` formats? How does that vary depending on the platform?